interfaces.go

```diff
@@ -34,15 +34,17 @@ type SpanRecorder interface {
 	RecordSpan(RawSpan)
 }
 
-// Tracer extends the opentracing.Tracer interface with methods to
-// probe implementation state, for use by basictracer consumers.
+// Tracer extends the `opentracing.Tracer` interface with methods for manual
+// flushing and closing. To access these methods, you can take the global
+// tracer and typecast it to a `lightstep.Tracer`. As a convenience, the
+// lightstep package provides static functions which perform the typecasting.
 type Tracer interface {
 	ot.Tracer
 
 	// Close flushes and then terminates the LightStep collector
-	Close() error
+	Close(context.Context)
```

Comment:
Just noticed this, adding context here is a breaking API change, right? Do we want to do that? Would it be better to add new methods (`CloseContext(...)` and `FlushContext(...)`). I have noticed that pattern in the core go code when they want to add context but don't want it to be backward incompatible.